.github/workflows/test.yml

```diff
@@ -0,0 +1,158 @@
+name: gha-workflow-pt-test
+on: [push, pull_request]
```

Comment:
We might to limit this to our main branch to avoid duplicate runs, but let's see. https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration